design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,183 @@
+
+![admin-actions](./obc-admin.png)
+
+These steps may be performed in any order.
+
+The admin ...
+
+- creates the ObjectBucketClaim Operator (a Deployment API object)
+- creates a StorageClass with `parameters[serviceName]=objectstore-service` and `parameters[serviceNamespace]=some-namespace`
+  - `provisioner` field is left blank.  Since only the ObjectBucketClaim operator is watching ObjectBucketClaim instances, defining the operator is unnecessary. 
+
+Once the operator is running, it will be begin watching ObjectBucketClaims.
+- An object store has been provisioned with a reachable endpoint.  ([e.g. Rook-Ceph Object](https://rook.github.io/docs/rook/master/ceph-object.html))
+
+#### MVP Use Cases
+
+**Use Case: Expose an Object Store Endpoint**
+
+_As a Kubernetes admin, I want to expose an object store to cluster users for bucket provisioning via the Kubernetes API._
+
+![admin-actions](./obc-admin.png)
+
+These next steps may be performed in any order.
+
+_The administrator ..._
+- creates the Rook ObjectBucketClaim Operator
+- creates a StorageClass with `parameters[objectStoreService]=in-cluster-service` and `parameters[objectStoreNamespace]=admin-namespace`
+
+
+
+**Use Case: Provision a Bucket** 
+
+_As a Kubernetes user, I want to leverage the Kubernetes API to create S3 buckets. I expect to get back the bucket connection information in a ConfigMap._
+ 
+![user-actions](./obc-user.png)
+
+1. The Rook Operator detects a new ObjectBucketClaim instance.  
+    1. The operator uses `objectBucketClaim.spec.secretName` to get the S3 access keys secret.  
+    1. The operator uses `objectBucketClaim.spec.storageClassName` to get the `Service` endpoint of the object store.
+1. The operator uses the object store endpoint and access keys for an S3 "make bucket" call.
+1. The operator creates a ConfigMap in the namespace of the ObjectBucketClaim with relevant connection data of the bucket.
+1. An app Pod may then mount the Secret and the ConfigMap to begin accessing the bucket. 
+
+**Use Case:** Delete an Object Bucket
+
+_As a Kubernetes user, I want to delete ObjectBucketClaim instances and cleanup generated API resources._
+
+1. The user deletes the ObjectBucketClaim via `kubectl delete ...`.
+1. The ObjectBucketClaim is marked for deletion and left in the foreground.
+1. The respective ConfigMap is deleted.
+1. The ObjectBucketClaim is garbage collected.
+
+---
+
+## API Specifications
+
+#### ObjectBucketClaim CRD
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1 
+kind: CustomResourceDefinition
+metadata:
+  name: objectbucketclaims.rook.io
+spec:
+  group: rook.io
+  version: v1alpha2
+  scope: namespaced
+  names:
+      kind: ObjectBucketClaim
+      listKind: ObjectBucketClaimList
+      singular: objectbucketclaim
+      plural: objectbucketclaims
+```
+
+#### ObjectBucketClaim
+
+```yaml
+apiVersion: rook.io/v1alpha2
+kind: ObjectBucketClaim
+metadata:
```

Comment:
should there be an `ownerReferences` value pointing to the ceph cluster here? IOW, when the cluster is deleted this CRD would also be deleted.